pkgs/build-support/go/module.nix

```diff
@@ -85,6 +87,7 @@ let
       runHook preConfigure
       export GOCACHE=$TMPDIR/go-cache
       export GOPATH="$TMPDIR/go"
+      export GOPROXY="''${GOPROXY:-direct}" # respect impureEnvVars
```

Comment:
Personally I think keeping it default (and making it overridable is better. Having sumdb backed checking seems like a win to me.

Reply:
> Personally I think keeping it default 

Do you mean the leaving it unset or setting `"https://proxy.golang.org,direct"`?

I'd prefer not to set it at but but need to do something to address https://github.com/NixOS/nixpkgs/pull/238100#issuecomment-1601932975.

> and making it overridable is better

Besides setting it via `impureEnvVars`?

Reply:
From my understanding of:
>  When GOTOOLCHAIN is set to local, the go command always runs the bundled Go toolchain.

We only need to set GOTOOLCHAIN to address the toolchain downloading concern.. Setting GOPROXY will impact module fetching as well as (if we didn't already have it as `local` toolchain fetching)

Reply:
What does GOTOOLCHAIN have to do with the GOPROXY error?

Reply:
oh, my bad, I clicked it and read the comment below ;D

In that case, I think setting it to `https://proxy.golang.org,direct` is the best approach (assuming it fixes the error).

Reply:
The `''` in the var value seems like a typo to me?

Reply:
> In that case, I think setting it to https://proxy.golang.org,direct is the best approach (assuming it fixes the error).

I expect people probably won't like that `proxy.golang.org` is set here even although it is the default that we've been using anyway.

> The `''` in the var value seems like a typo to me?

It isn't a typo, need to escape `${}`.

Reply:
The `GOPROXY` issue was fixed in https://github.com/NixOS/nixpkgs/commit/cbc976a97c3372e1eec5db021db994b85e098d12 so I'll revert this in https://github.com/NixOS/nixpkgs/pull/250492.